### PR TITLE
Fix compilation issues

### PR DESCRIPTION
**Pull request by archey**
Fixes a few compilation warnings with Glibc >=2.20 and gcc 5.3

### CODE REVIEW

File: astraceroute.c [hunk 1]
```diff
@@ -4,7 +4,9 @@
  * Subject to the GPL, version 2.
  */
 
-#define _BSD_SOURCE
+#ifdef _BSD_SOURCE
+# define _DEFAULT_SOURCE
+#endif
```
Comment (tklauser) [inline astraceroute.c]: Judging from the documentation at http://man7.org/linux/man-pages/man7/feature_test_macros.7.html it seems we should define both, _BSD_SOURCE and _DEFAULT regardless of any existing definitions, i.e.

`#define _BSD_SOURCE`
`#define _DEFAULT_SOURCE`

Does this fix the warnings as well (sorry, I don't have a system with glibc >= 2.20 available right now)? If yes, could you please update this patch (just force push the updated commit to your branch). Thanks!

File: tprintf.c [hunk 2]
```diff
@@ -5,7 +5,9 @@
  * Subject to the GPL, version 2.
  */
 
-#define _BSD_SOURCE
+#ifdef _BSD_SOURCE
+# define _DEFAULT_SOURCE
+#endif
```
Comment (tklauser) [inline tprintf.c]: Same remark here, see comment above.

**Comment by tklauser** [comment]
The _GNU_SOURCE patch looks good to me. As for the _BSD_SOURCE patch, please see my line notes.

**Comment by archey** [comment]
@tklauser Yes I believe your correct if we define _BSD_SOURCE we have to define _DEFAULT_SOURCE. I pushed the changes to my branch to account for this and it compiles cleanly on glibc>=2.20.

**Comment by tklauser** [comment]
Thanks for the updated patch. I took the liberty to squash the two `_BSD_SOURCE` together into one. Merged as a0f448ce and f5ae2cd9.

**Pull request closed without merging**